src/createBlock.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Creates the next block in the Raha Blockchain from unapplied operations in Firestore.
+ *
+ * Example usage:
+ * node dist/createBlock.js > newBlock.json
+ */
+
+import { getBlockchain } from './RahaBlockchain';
+import { operationsCollectionFilters as filters, get, operationsCollection } from './RahaFirestore';
+import { BLOCKCHAIN_VERSION_NO, Block, StellarMetadata, IpfsBlock, Operation, RequestInvite, Trust } from './schema';
+
+/**
+ * A map of functions that build the data component of Blockchain Operations from
+ * the data component of Firestore Operations.
+ */
+const bcOpDataBuilders = {
+    'REQUEST_INVITE': (fsOpData): RequestInvite => ({
+        full_name: fsOpData.full_name,
+        to_mid: fsOpData.to_mid,
+        video_url: fsOpData.video_url,
+    }),
+    'TRUST': (operationData): Trust => ({
+        to_mid: operationData.to_mid,
+    }),
+}
+
+/**
+ * Filter new operations are invalid within the context of the current blockchain.
+ * This means:
+ *   * users cannot perform operations on nonexistent users.
+ *   * // TODO: there must exist a trust path from a user who joined in a previous block to any user who performs a new TRUST operation
+ */
+function filterBcOpsAgainstBlockchain(blockchain, bcOps) {
+    // Create a set of creator_mid on all REQUEST_INVITE operations in blockchain and bcOps.
+    // Check if to_mid, from_mid is included in above set.
+    // TODO;
+    return bcOps;
+}
+
+/**
+ * TODO.
+ */
+function validateBcOp(bcOp) {
+    return true;
+}
+
+/**
+ * Build a Blockchain Operation from a Firestore Operation. Returns undefined if the operation is invalid.
+ */
+function buildBcOpFromFsOp(
+        fsOp, index: number): Operation|void {
```

Comment:
I think the proper return type is `Operation | undefined`

Also I usually format something like this as

```
function buildBcOpFromFsOp(
  fsOp, index: number
): Operation | undefined {

}
```

just so that the auto tabbing in most editors works consistently, no big hanging indents, etc.

Additionally, would be nice if you explicitly typed the `fsOp` parameter, though i assume you find that annoying given that `noImplicitAny` is turned off in the `tsconfig`?